pkg/plugins/manager/loader/loader.go

```diff
@@ -195,6 +198,14 @@ func (l *Loader) loadPlugins(ctx context.Context, class plugins.Class, pluginJSO
 			return nil, err
 		}
 		metrics.SetPluginBuildInformation(p.ID, string(p.Type), p.Info.Version, string(p.Signature))
+
+		if errDeclareRoles := l.roleRegistry.DeclarePluginRoles(ctx, p.ID, p.Name, p.Roles); errDeclareRoles != nil {
+			l.log.Warn("Declare plugin roles failed",
+				"pluginID", p.ID,
+				"warning", "Make sure the role declaration is correct.",
+				"path", p.PluginDir+"/plugin.json",
```

Comment:
Nit. Maybe don't need to append plugin.json since `p.PluginDir` should provide enough context?